presto-product-tests/src/main/java/com/facebook/presto/tests/SqlCancelTests.java

```diff
@@ -0,0 +1,196 @@
+                return true;
+            }
+        }
+    }
+
+    private class SqlCancelCallable
+            implements Callable<Response>
+    {
+        private String sqlStatement;
+
+        public SqlCancelCallable(String sqlStatement)
+        {
+            this.sqlStatement = sqlStatement;
+        }
+
+        public Response call()
+        {
+            long endTime = System.currentTimeMillis() + WAIT_TIME;
+            while (System.currentTimeMillis() < endTime && !Thread.currentThread().isInterrupted()) {
+                QueryResult queryResult = query(format("SELECT state, query_id from system.runtime.queries" +
+                        " WHERE query = '%s' ORDER BY created DESC LIMIT 1", sqlStatement));
+                if (queryResult.getRowsCount() == 0) {
+                    sleepUninterruptibly(STEP, TimeUnit.MILLISECONDS);
+                    continue;
+                }
+                String queryState = (String) queryResult.row(0).get(0);
+                String queryId = (String) queryResult.row(0).get(1);
+                if (!queryState.equalsIgnoreCase("running")) {
+                    sleepUninterruptibly(STEP, TimeUnit.MILLISECONDS);
+                }
+                else {
+                    return queryCanceller.cancel(queryId, serverAddress);
+                }
+            }
+            return null;
+        }
+    }
+
+    private void doCancelTest(String sql)
+            throws Exception
+    {
+        Future<Boolean> executeQueryThread = executor.submit(new SqlExecuteCallable(sql));
+        Future<Response> cancelQueryThread = executor.submit(new SqlCancelCallable(sql));
+
+        boolean isQueryCancelled;
+        try {
+            isQueryCancelled = executeQueryThread.get(2, TimeUnit.MINUTES);
+        }
+        catch (TimeoutException e) {
+            executeQueryThread.cancel(true);
+            throw new RuntimeException("Query execution time limit exceeded", e);
+        }
+
+        String queryState = (String) query(format("SELECT state from system.runtime.queries" +
+                " WHERE query = '%s' ORDER BY created DESC LIMIT 1", sql)).row(0).get(0);
+        assertEquals(queryState.equalsIgnoreCase("failed"), true, "Failed to cancel query");
+        assertEquals(isQueryCancelled, true, "Failed to cancel query");
+
+        Response response = cancelQueryThread.get();
+        assertNotNull(response, "Error in cancel query thread");
+
+        if (response.getStatusCode() != HttpStatus.OK.code() && response.getStatusCode() != HttpStatus.ACCEPTED.code() &&
+                response.getStatusCode() != HttpStatus.NO_CONTENT.code()) {
+            fail("Unexpected error code " + response.getStatusCode() + "; reason=" + response.getStatusMessage());
+        }
+    }
+
+    @Test(groups = {CANCEL_QUERY})
+    public void cancelCreateTableTest()
+            throws Exception
+    {
+        String sql = format("CREATE TABLE %s AS " +
```

Comment:
This is not needed because I get the queryId of the latest query executed using "ORDER BY created" clause. Also, all the tests have different queries.